### PR TITLE
feat(autoware_cmake): set system includes after ament_auto_find_build_dependencies

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

A follow-up for #76.

There are still some errors left after merging https://github.com/autowarefoundation/autoware.universe/pull/978.
https://github.com/autowarefoundation/autoware/runs/6614506073?check_suite_focus=true#step:6:5365

Since it's hard to check all packages, I'd like to add Eigen as a system include here.
As I removed `find_package(Eigen3)` from #76, I believe this doesn't add any unnecessary dependency that Esteve-san was concerned about.

### Example

With a debug message added like this,

```
  message("#### ${EIGEN3_INCLUDE_DIR}")
  include_directories(SYSTEM
    ${EIGEN3_INCLUDE_DIR}
  )
```

- interpolation (Indirect Eigen dependency through tier4_autoware_utils)

```
Starting >>> interpolation
--- stderr: interpolation                                
#### /usr/include/eigen3
---
Finished <<< interpolation [13.5s]
```

- system_monitor (No Eigen dependency)

```
Starting >>> system_monitor
--- stderr: system_monitor                               
#### 
---
Finished <<< system_monitor [2.92s]
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kenji-miyake** [comment]
CI: https://github.com/autowarefoundation/autoware/actions/runs/2394217062

File: autoware_cmake/cmake/autoware_package.cmake [hunk 1]
```diff
@@ -48,6 +48,11 @@ macro(autoware_package)
   find_package(ament_cmake_auto REQUIRED)
   ament_auto_find_build_dependencies()
 
+  # Set common system includes
+  include_directories(SYSTEM
+    ${EIGEN3_INCLUDE_DIR}
+  )
+
```
Comment (xmfcx) [inline autoware_cmake/cmake/autoware_package.cmake:55]: @esteve let's add this and create we can open up a PR to remove it by adding it to necessary packages later on, what do you think?

**Review by xmfcx — APPROVED** [review]

**Comment by kenji-miyake** [comment]
I'll wait for @esteve -san's review.

File: autoware_cmake/cmake/autoware_package.cmake (hunk 1 above)
Comment (esteve) [reply autoware_cmake/cmake/autoware_package.cmake:55]: @xmfcx sounds good to me

**Review by esteve — APPROVED** [review]

**Comment by kenji-miyake** [comment]
Thank you for your reviews and approvals! :smile:

**Pull request merged**